src/Workspaces/Core/Portable/Utilities/Operations/IOperationScope.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.Utilities
+{
+    /// <summary>
+    /// Represents a single scope of a context of executing potentially long running operation. Scopes allow multiple
+    /// components running within an operation to share the same context.
+    /// </summary>
+    public interface IOperationScope : IDisposable
```

Comment:
equivalent to IUIThreadOperationScope.   As above, users cannot disable cancellation.  They can also not 'take ownership'.